lib/validate.js

```diff
@@ -81,6 +120,29 @@ module.exports = function validateMiddleware (opts) {
               msg: 'Bad Request - ' + err.toString()
             });
           } else {
+            // Assemble object from properties deemed valid from
+            // all Joi schema validations.
+            var validParams = _.assign.apply(_, joiedParams);
+
+            // See if we removed any params, if so, log it.
+            var removedParams = _.difference(_.keys(data), _.keys(validParams));
+            if(removedParams.length > 0) {
+              log.info(
+                '%s params removed for "%s" during joi validation: %s. %s %s',
+                (type === 'list') ? 'req.query' : 'req.body',
+                type,
+                removedParams,
```

Comment:
Could also do this "%s params removed for "%s" during joi validation: %s. %s" and pass _removedParams_ as the last argument instead. This will cause bunyan to auto parse it, and also it will parse it safely, whereas using "%j" can throw (pretty odd behaviour, right?)